proto/auction/v1/auction.proto

```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
+package auction.v1;
+
+import "cosmos/base/v1beta1/coin.proto";
+import "cosmos_proto/cosmos.proto";
+import "gogoproto/gogo.proto";
+
+option go_package = "github.com/peggyjv/sommelier/v4/x/auction/types";
+
+message Auction {
+  uint32 id = 1;
+  cosmos.base.v1beta1.Coin starting_tokens_for_sale = 2 [ (gogoproto.nullable) = false ];
+  uint64 start_block = 3;
+  uint64 end_block = 4;
+  string initial_price_decrease_rate = 5 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
+  string current_price_decrease_rate = 6 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
+  uint64 price_decrease_block_interval = 7;
+  string initial_unit_price_in_usomm = 8 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
+  string current_unit_price_in_usomm = 9 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
+  cosmos.base.v1beta1.Coin remaining_tokens_for_sale = 10 [ (gogoproto.nullable) = false ];
+  string funding_module_account = 11;
+  string proceeds_module_account = 12;
+}
+
+message Bid {
+  uint64 id = 1;
+  uint32 auction_id = 2;
+  string bidder = 3;
+  cosmos.base.v1beta1.Coin max_bid_in_usomm = 4 [ (gogoproto.nullable) = false ];
+  cosmos.base.v1beta1.Coin sale_token_minimum_amount = 5 [ (gogoproto.nullable) = false ];
+  cosmos.base.v1beta1.Coin total_fulfilled_sale_tokens = 6 [ (gogoproto.nullable) = false ];
+  string sale_token_unit_price_in_usomm = 7 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
+  cosmos.base.v1beta1.Coin total_usomm_paid = 8 [
+    (gogoproto.nullable) = false
+  ];
+}
+
+message TokenPrice {
+  string denom = 1;
+  string usd_price = 2 [
```

Comment:
Should we price everything in dollars or in some?

Reply:
The thought process behind pricing everything in dollars is that it requires less calculations on the part of the proposer as well as less calculations on the part of the ones verifying the proposal to make sure all the price conversions into usomm are accurate. 

With this approach all the calculations for pricing sale tokens in terms of usomm are done at auction start time and can be trusted implicitly as long as the submitted usd prices are accurate.